source/common/network/connection_impl.cc

```diff
@@ -224,6 +224,11 @@ void ConnectionImpl::enableHalfClose(bool enabled) {
 
 void ConnectionImpl::readDisable(bool disable) {
   ASSERT(state() == State::Open);
+  if (file_event_ == nullptr) {
```

Comment:
Can this check be "if (state() != State::Open)"? Does that end up covering the same case? 

Also, maybe a comment saying that this is a short-term fix. Because normally, when I read an ASSERT in the code, I assume this case can't happen, but that's not true here, so an explanation of what's going on would help.

Reply:
How about both?  I believe we shouldn't run this code if we're either closing or closed.   I'd feel more comfortable explicitly checking for the nullptr which will cause the crash though, just in case someone ever mistakenly inserts a flush (or other code) between resetting the file_event_ (making this function unsafe) and closing the underlying socket (making state != Open)

W.r.t. comments, I don't think we should take this code out.  I think this is long term error handling for any future broken code of this type.